docs/src/proposals/traced-transaction-construction.md

```diff
@@ -0,0 +1,247 @@
+# Traced transaction construction via RPC's `simulateTransaction`
+
+## Problem
+
+With more composition comes in the Solana's program ecosystem, it will be harder
```

Comment:
this doc is still in the draft phase. but I think the first half of this is written decently. ;)